bin/fast_composition_analyzer.py

```diff
@@ -0,0 +1,520 @@
+                left_repeat = pred.left_repeat
+                right_repeat = pred.right_repeat
+
+                right_repeat_sequence = str(right_repeat.sequence)
+                if right_repeat.type == 'inverted':
+                    right_repeat_sequence = Seq(right_repeat_sequence).reverse_complement()
+
+                # Create prediction name
+                pred_name = f"{outlier.contig}_{i:03d}"
+                outlier_coords = f"{left_repeat.start}-{right_repeat.end}"
+                
+                # Score information
+                score_info = (f"gc:{outlier.gc_score:.2f}|"
+                            f"kmer:{outlier.kmer_score:.2f}|"
+                            f"combined:{outlier.combined_score:.2f}")
+                
+                confidence_info = f"conf:{outlier.confidence:.3f}"
+                
+                # Write left repeat
+                left_info = (f"{pred_name}:{outlier_coords}|"
+                           f"{left_repeat.type}_1|{left_repeat.sequence}|"
+                           f"{score_info}|{confidence_info}")
+                f.write(f"{outlier.contig}\t{left_repeat.start}\t{left_repeat.end}\t{left_info}\n")
+                
+                # Write right repeat
+                right_info = (f"{pred_name}:{outlier_coords}|"
+                            f"{right_repeat.type}_2|{right_repeat_sequence}|"
+                            f"{score_info}|{confidence_info}")
+                f.write(f"{outlier.contig}\t{right_repeat.start}\t{right_repeat.end}\t{right_info}\n")
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Fast compositional outlier detection in long contigs",
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        "--input_fasta",
+        help="Input FASTA file with contigs"
+    )
+    parser.add_argument(
+        "--output_bed",
+        help="Output BED file with predictions"
+    )
+    parser.add_argument(
+        "--score-threshold",
+        type=float,
+        default=2.0,
+        help="Z-score threshold for outlier detection (default: 2.0)"
+    )
+    parser.add_argument(
+        "--min-contig-length",
+        type=int,
+        default=100000,
+        help="Minimum contig length to consider (default: 100000)"
+    )
+    parser.add_argument(
+        "--window-size",
+        type=int,
+        default=5000,
+        help="Window size for compositional analysis (default: 5000)"
+    )
+    parser.add_argument(
+        "--threads",
+        type=int,
+        help="Number of threads to use (default: auto-detect, max 4)"
+    )
+    args = parser.parse_args()
+    
+    # Running the pipeline
+    pipeline = FastGenomicOutlierPipeline(
+        min_contig_length=int(args.min_contig_length),
+        window_size=int(args.window_size),
+        step_size=int(args.window_size/2),
+        score_threshold=float(args.score_threshold),
+        n_threads=int(args.threads)
```

Comment:
No need to do `int(args.xxx)` if you set the type as int when defining the argument